monitoring/ceph-mixin/prometheus_alerts.libsonnet

```diff
@@ -0,0 +1,695 @@
+          labels: { severity: 'warning', type: 'ceph_default', oid: '1.3.6.1.4.1.50495.1.2.1.4.2' },
+          annotations: {
+            documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#osd-down',
+            summary: 'An OSD has been marked down/unavailable%(cluster)s' % $.MultiClusterSummary(),
+            description: |||
+              {{ $num := query "count(ceph_osd_up == 0)" | first | value }}{{ $s := "" }}{{ if gt $num 1.0 }}{{ $s = "s" }}{{ end }}{{ $num }} OSD{{ $s }} down for over 5mins.  The following OSD{{ $s }} {{ if eq $s "" }}is{{ else }}are{{ end }} down:   {{- range query "(ceph_osd_up * on(ceph_daemon) group_left(hostname) ceph_osd_metadata) == 0"}}   - {{ .Labels.ceph_daemon }} on {{ .Labels.hostname }}   {{- end }}
+            |||,
+          },
+        },
+        {
+          alert: 'CephOSDNearFull',
+          'for': '5m',
+          expr: 'ceph_health_detail{name="OSD_NEARFULL"} == 1',
+          labels: { severity: 'warning', type: 'ceph_default', oid: '1.3.6.1.4.1.50495.1.2.1.4.3' },
+          annotations: {
+            documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#osd-nearfull',
+            summary: 'OSD(s) running low on free space (NEARFULL)%(cluster)s' % $.MultiClusterSummary(),
+            description: "One or more OSDs have reached their NEARFULL threshold  Use 'ceph health detail' to identify which OSDs have reached this threshold. To resolve, either add capacity to the cluster, or delete unwanted data ",
+          },
+        },
+        {
+          alert: 'CephOSDFull',
+          'for': '1m',
+          expr: 'ceph_health_detail{name="OSD_FULL"} > 0',
+          labels: { severity: 'critical', type: 'ceph_default', oid: '1.3.6.1.4.1.50495.1.2.1.4.6' },
+          annotations: {
+            documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#osd-full',
+            summary: 'OSD(s) is full, writes blocked%(cluster)s' % $.MultiClusterSummary(),
+            description: "An OSD has reached it's full threshold. Writes from all pools that share the affected OSD will be blocked.  To resolve, either add capacity to the cluster, or delete unwanted data ",
+          },
+        },
+        {
+          alert: 'CephOSDBackfillFull',
+          'for': '1m',
+          expr: 'ceph_health_detail{name="OSD_BACKFILLFULL"} > 0',
+          labels: { severity: 'warning', type: 'ceph_default' },
+          annotations: {
+            documentation: 'https://docs.ceph.com/en/latest/rados/operations/health-checks#osd-backfillfull',
+            summary: 'OSD(s) too full for backfill operations%(cluster)s' % $.MultiClusterSummary(),
+            description: "An OSD has reached it's BACKFILL FULL threshold. This will prevent rebalance operations completing for some pools. Check the current capacity utilisation with 'ceph df'  To resolve, either add capacity to the cluster, or delete unwanted data ",
```

Comment:
s/it's/the/
This will prevent rebalancing and recovery completion.  Check capacity utilization with `ceph df` and `ceph osd df`.  To resolve, add capacity to the cluster, delete unwanted data, bring back down or destroyed OSDs, and / or check operation of the balancer module"